src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs

```diff
@@ -290,6 +290,38 @@ private static void AddPopParams(AuthenticationRequestParameters authenticationR
 
             return msalTokenResponse;
         }
+        
+        /// <summary>
+        /// Token Received from WAM
+        /// </summary>
+        /// <param name="authResult"></param>
+        /// <param name="logger"></param>
+        private static bool TokenReceivedFromWam(
+            NativeInterop.AuthResult authResult,
+            ILoggerAdapter logger)
+        {
+            //success result from WAM
+            if (authResult.IsSuccess)
+                return true;
+            
+            //user switch result is not success and a token is received from MSALRuntime with an Error Response status    
+            if (authResult.Error != null 
+                && (ResponseStatus)authResult.Error.Status == ResponseStatus.UserSwitch)
```

Comment:
Great fix - "UserSwitch" (i.e. when the user chooses "Use another account" in the UI) is a success case from MSALs point of view. 

CC @siddhijain @rayluo @tnorling

Reply:
Created a backlog item to take care of this in Java -https://identitydivision.visualstudio.com/Engineering/_boards/board/t/Auth%20Client%20-%20JAVA/Backlog%20items/?showParents=true&workitem=2398992

Reply:
@rayluo - do you know if MSAL Python treats this as success?